python/tk_framework_desktopserver/server_protocol.py

```diff
@@ -242,30 +243,33 @@ def _validate_user(self, user_id):
         # origin is formatted such as https://xyz.shotgunstudio.com:port_number
         # host is https://xyz.shotgunstudio.com:port_number
         host_network = self.factory.host.lower()
-        origin_network = self._origin.lower()
+        parsed_host = urlparse.urlparse(self._origin)
+        origin_network = (parsed_host.hostname or parsed_host.netloc).lower()
```

Comment:
nitpick: probably good to include the same comment here as above where you also did the fallback to netloc.